src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -58,5 +58,41 @@ public void AddOrUpdatePackageSourceMappingSourceItem(PackageSourceMappingSource
 
             _settings.SaveToDisk();
         }
+
+        public void SavePackageSourceMappings(IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems)
+        {
+            if (packageSourceMappingsSourceItems == null)
+            {
+                throw new ArgumentNullException(nameof(packageSourceMappingsSourceItems));
+            }
+
+
+            var existingSettingsLookup = GetPackageSourceMappingItems();
+
+            foreach (var sourceMappingItem in packageSourceMappingsSourceItems)
+            {
+                if (existingSettingsLookup != null)
+                {
+                    AddOrUpdatePackageSourceMappingSourceItem(sourceMappingItem);
+                }
+            }
+
+            //Remove all old mappings not in new mappings
+            if (existingSettingsLookup != null)
+            {
+                List<PackageSourceMappingSourceItem> removeMappings = new List<PackageSourceMappingSourceItem>();
+                foreach (var sourceItem in existingSettingsLookup)
+                {
+                    if (!packageSourceMappingsSourceItems.Contains(sourceItem))
+                    {
+                        removeMappings.Add(sourceItem);
+                    }
+                }
+                if (removeMappings != null && removeMappings.Count > 0)
```

Comment:
A little bit better perf.

```suggestion
                if (removeMappings != null && removeMappings.Any())
```

Reply:
This suggestion should not be taken as using LINQ's `.Any()` is slower than `.Count >0` (and depending on implementation might require allocations, whereas checking Count should never need allocations.